implementations/rust/vault/src/c/mod.rs

```diff
@@ -1,20 +1,771 @@
-use crate::error::VaultFailError;
-use crate::types::{PublicKey, SecretKey, SecretKeyAttributes, SecretKeyContext};
+use crate::error::{VaultFailError, VaultFailErrorKind};
+use crate::types::{
+    PublicKey, SecretKey, SecretKeyAttributes, SecretKeyType, SecretPersistenceType,
+    SecretPurposeType,
+};
+use crate::{Secret, Vault};
 use c_bindings::*;
-use std::cell::Cell;
+use c_rust_memory::RustAlloc;
+use failure::_core::any::Any;
+use std::ffi::CStr;
 use zeroize::Zeroize;
 
+/// C vault secret
+#[derive(Debug)]
+pub struct CVaultSecret {
+    handle: ockam_vault_secret_t,
+}
+
+// FIXME
+unsafe impl Send for CVaultSecret {}
```

Comment:
Thoughts?

Reply:
Could this be changed to just be a pointer directly rather than wrapping the secret_t? In other words, could we just use *mut ockam_vault_secret_t?

Reply:
Maybe yes, but I don't like the idea of populating C structures all over the place. Native Rust struct that hides all this nasty details under the hood is IMHO better